File: ios/ProfileFieldMapper.swift
import Octopus

struct ProfileFieldMapper {
  static func fromReactNativeArray(_ appManagedFields: [String]?) -> Set<ConnectionMode.SSOConfiguration.ProfileField> {
    guard let fields = appManagedFields else {
      return []
    }

    var profileFields: Set<ConnectionMode.SSOConfiguration.ProfileField> = []
    for field in fields {
      if let profileField = fromReactNativeString(field) {
        profileFields.insert(profileField)
      }
    }
    return profileFields
  }

  static func fromReactNativeString(_ fieldName: String?) -> ConnectionMode.SSOConfiguration.ProfileField? {
    switch fieldName {
    case "username":
      return .nickname
    case "biography":
      return .bio
    case "profilePicture":
      return .picture
    default:
      return nil
    }
  }

  static func toReactNativeString(_ profileField: ConnectionMode.SSOConfiguration.ProfileField?) -> String? {
    switch profileField {
    case .nickname:
      return "username"
    case .bio:
      return "biography"
    case .picture:
      return "profilePicture"
    default:
      return nil
    }
  }
}

File: android/src/main/java/com/octopuscommunity/octopusreactnativesdk/ProfileFieldMapper.kt
package com.octopuscommunity.octopusreactnativesdk

import com.facebook.react.bridge.ReadableArray
import com.octopuscommunity.sdk.domain.model.ProfileField

object ProfileFieldMapper {

  fun fromReactNativeArray(appManagedFields: ReadableArray?): Set<ProfileField> {
    if (appManagedFields == null) {
      return emptySet()
    }

    val profileFields = mutableSetOf<ProfileField>()
    for (i in 0 until appManagedFields.size()) {
      fromReactNativeString(appManagedFields.getString(i))?.let { profileField ->
        profileFields.add(profileField)
      }
    }
    return profileFields
  }

  fun fromReactNativeString(fieldName: String?): ProfileField? {
    return when (fieldName) {
      "username" -> ProfileField.NICKNAME
      "biography" -> ProfileField.BIO
      "profilePicture" -> ProfileField.AVATAR
      else -> null
    }
  }

  fun toReactNativeString(profileField: ProfileField?): String? {
    return when (profileField) {
      ProfileField.NICKNAME -> "username"
      ProfileField.BIO -> "biography"
      ProfileField.AVATAR -> "profilePicture"
      else -> null
    }
  }
}
